make fails

As reported by @dandv
```
$ make
cc -O3 -Wall -g -c alt.c -o alt.o
cc -O3 -Wall -g -c altlist.c -o altlist.o
cc -O3 -Wall -g -c build_structs.c -o build_structs.o
build_structs.c: In function ‘buildstructs_is_useless_perm’:
build_structs.c:259:9: warning: variable ‘alt_id’ set but not used [-Wunused-but-set-variable]
     int alt_id = 0;
         ^
cc -O3 -Wall -g -c char_set.c -o char_set.o
cc -O3 -Wall -g -c data.c -o data.o
cc -O3 -Wall -g -c debug.c -o debug.o
cc -O3 -Wall -g -c grouplist.c -o grouplist.o
cc -O3 -Wall -g -c grouprecord.c -o grouprecord.o
cc -O3 -Wall -g -c main.c -o main.o
cc -O3 -Wall -g -c memory.c -o memory.o
cc -O3 -Wall -g -c parse_regex.c -o parse_regex.o
cc -O3 -Wall -g -c program_args.c -o program_args.o
program_args.c: In function ‘program_args_parse’:
program_args.c:268:41: warning: passing argument 2 of ‘parse_regex_pass_char_class’ makes pointer from integer without a cast [-Wint-conversion]
   parse_regex_pass_char_class(g->regex, g->debug_code & D_Program_Args & D_Parse_Regex_Eachstep);
                                         ^
In file included from program_args.c:28:0:
parse_regex.h:55:6: note: expected ‘tnode {aka struct _tnode *}’ but argument is of type ‘int’
 void parse_regex_pass_char_class (char_set, tnode);
      ^
program_args.c:325:41: warning: passing argument 2 of ‘parse_regex_pass_char_class’ makes pointer from integer without a cast [-Wint-conversion]
   parse_regex_pass_char_class(g->regex, g->debug_code & D_Program_Args & D_Parse_Regex_Eachstep);
                                         ^
In file included from program_args.c:28:0:
parse_regex.h:55:6: note: expected ‘tnode {aka struct _tnode *}’ but argument is of type ‘int’
 void parse_regex_pass_char_class (char_set, tnode);
      ^
cc -O3 -Wall -g -c re_perm.c -o re_perm.o
cc -O3 -Wall -g -c silist.c -o silist.o
cc -O3 -Wall -g -c tnode.c -o tnode.o
cc -O3 -Wall -g -c vlr.c -o vlr.o
cc -O3 -Wall -g -c vlrlist.c -o vlrlist.o
gcc -O3 -Wall -g -lm alt.o altlist.o build_structs.o char_set.o data.o \
debug.o grouplist.o grouprecord.o main.o memory.o \
parse_regex.o program_args.o re_perm.o silist.o tnode.o \
vlr.o vlrlist.o -o regldg
char_set.o: In function `char_set_g_add_universe':
/c/prg/regldg-1.0.0/regldg/char_set.c:109: undefined reference to `pow'
/c/prg/regldg-1.0.0/regldg/char_set.c:111: undefined reference to `pow'
char_set.o: In function `char_set_g_display':
/c/prg/regldg-1.0.0/regldg/char_set.c:143: undefined reference to `ceil'
debug.o: In function `debug_print':
/c/prg/regldg-1.0.0/regldg/debug.c:43: undefined reference to `log'
/c/prg/regldg-1.0.0/regldg/debug.c:43: undefined reference to `rint'
parse_regex.o: In function `parse_regex_print_group':
/c/prg/regldg-1.0.0/regldg/parse_regex.c:392: undefined reference to `log10'
parse_regex.o: In function `parse_regex_print_char_class':
/c/prg/regldg-1.0.0/regldg/parse_regex.c:1113: undefined reference to `log10'
parse_regex.o: In function `parse_regex_read_dec_val':
/c/prg/regldg-1.0.0/regldg/parse_regex.c:1438: undefined reference to `pow'
parse_regex.o: In function `parse_regex_pass_universe_specifier':
/c/prg/regldg-1.0.0/regldg/parse_regex.c:1261: undefined reference to `log10'
parse_regex.o: In function `parse_regex_scan':
/c/prg/regldg-1.0.0/regldg/parse_regex.c:241: undefined reference to `log10'
/c/prg/regldg-1.0.0/regldg/parse_regex.c:192: undefined reference to `log10'
collect2: error: ld returned 1 exit status
Makefile:21: recipe for target 'all' failed
make: *** [all] Error 1
```

@dandv What hardware and OS are you running on, and what compiler are you using?

On OS X, I get:
```
wana:regldg pcronin$ make
cc -O3 -Wall -g -c alt.c -o alt.o
cc -O3 -Wall -g -c altlist.c -o altlist.o
cc -O3 -Wall -g -c build_structs.c -o build_structs.o
cc -O3 -Wall -g -c char_set.c -o char_set.o
cc -O3 -Wall -g -c data.c -o data.o
cc -O3 -Wall -g -c debug.c -o debug.o
cc -O3 -Wall -g -c grouplist.c -o grouplist.o
cc -O3 -Wall -g -c grouprecord.c -o grouprecord.o
cc -O3 -Wall -g -c main.c -o main.o
cc -O3 -Wall -g -c memory.c -o memory.o
cc -O3 -Wall -g -c parse_regex.c -o parse_regex.o
cc -O3 -Wall -g -c program_args.c -o program_args.o
program_args.c:268:41: warning: incompatible integer to pointer conversion
      passing 'int' to parameter of type 'tnode' (aka 'struct _tnode *')
      [-Wint-conversion]
  ...g->debug_code & D_Program_Args & D_Parse_Regex_Eachstep);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./parse_regex.h:55:50: note: passing argument to parameter here
void parse_regex_pass_char_class (char_set, tnode);
                                                 ^
program_args.c:325:41: warning: incompatible integer to pointer conversion
      passing 'int' to parameter of type 'tnode' (aka 'struct _tnode *')
      [-Wint-conversion]
  ...g->debug_code & D_Program_Args & D_Parse_Regex_Eachstep);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./parse_regex.h:55:50: note: passing argument to parameter here
void parse_regex_pass_char_class (char_set, tnode);
                                                 ^
2 warnings generated.
cc -O3 -Wall -g -c re_perm.c -o re_perm.o
cc -O3 -Wall -g -c silist.c -o silist.o
cc -O3 -Wall -g -c tnode.c -o tnode.o
cc -O3 -Wall -g -c vlr.c -o vlr.o
cc -O3 -Wall -g -c vlrlist.c -o vlrlist.o
gcc -O3 -Wall -g -lm alt.o altlist.o build_structs.o char_set.o data.o \
	debug.o grouplist.o grouprecord.o main.o memory.o \
	parse_regex.o program_args.o re_perm.o silist.o tnode.o \
	vlr.o vlrlist.o -o regldg
gwana:regldg pcronin$
```

I got myself a copy of `Linux ubuntu 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux` and can verify the issue with `make`.

```
pcronin@ubuntu:~/src/regldg$ make
cc -O3 -Wall -g -c alt.c -o alt.o
cc -O3 -Wall -g -c altlist.c -o altlist.o
cc -O3 -Wall -g -c build_structs.c -o build_structs.o
build_structs.c: In function ‘buildstructs_is_useless_perm’:
build_structs.c:259:9: warning: variable ‘alt_id’ set but not used [-Wunused-but-set-variable]
     int alt_id = 0;
         ^
cc -O3 -Wall -g -c char_set.c -o char_set.o
cc -O3 -Wall -g -c data.c -o data.o
cc -O3 -Wall -g -c debug.c -o debug.o
cc -O3 -Wall -g -c grouplist.c -o grouplist.o
cc -O3 -Wall -g -c grouprecord.c -o grouprecord.o
cc -O3 -Wall -g -c main.c -o main.o
cc -O3 -Wall -g -c memory.c -o memory.o
cc -O3 -Wall -g -c parse_regex.c -o parse_regex.o
cc -O3 -Wall -g -c program_args.c -o program_args.o
program_args.c: In function ‘program_args_parse’:
program_args.c:268:41: warning: passing argument 2 of ‘parse_regex_pass_char_class’ makes pointer from integer without a cast [-Wint-conversion]
   parse_regex_pass_char_class(g->regex, g->debug_code & D_Program_Args & D_Parse_Regex_Eachstep);
                                         ^
In file included from program_args.c:28:0:
parse_regex.h:55:6: note: expected ‘tnode {aka struct _tnode *}’ but argument is of type ‘int’
 void parse_regex_pass_char_class (char_set, tnode);
      ^
program_args.c:325:41: warning: passing argument 2 of ‘parse_regex_pass_char_class’ makes pointer from integer without a cast [-Wint-conversion]
   parse_regex_pass_char_class(g->regex, g->debug_code & D_Program_Args & D_Parse_Regex_Eachstep);
                                         ^
In file included from program_args.c:28:0:
parse_regex.h:55:6: note: expected ‘tnode {aka struct _tnode *}’ but argument is of type ‘int’
 void parse_regex_pass_char_class (char_set, tnode);
      ^
cc -O3 -Wall -g -c re_perm.c -o re_perm.o
cc -O3 -Wall -g -c silist.c -o silist.o
cc -O3 -Wall -g -c tnode.c -o tnode.o
cc -O3 -Wall -g -c vlr.c -o vlr.o
cc -O3 -Wall -g -c vlrlist.c -o vlrlist.o
gcc -O3 -Wall -g -lm alt.o altlist.o build_structs.o char_set.o data.o \
debug.o grouplist.o grouprecord.o main.o memory.o \
parse_regex.o program_args.o re_perm.o silist.o tnode.o \
vlr.o vlrlist.o -o regldg
char_set.o: In function `char_set_g_add_universe':
/home/pcronin/src/regldg/char_set.c:109: undefined reference to `pow'
/home/pcronin/src/regldg/char_set.c:111: undefined reference to `pow'
char_set.o: In function `char_set_g_display':
/home/pcronin/src/regldg/char_set.c:143: undefined reference to `ceil'
debug.o: In function `debug_print':
/home/pcronin/src/regldg/debug.c:43: undefined reference to `log'
/home/pcronin/src/regldg/debug.c:43: undefined reference to `rint'
parse_regex.o: In function `parse_regex_print_group':
/home/pcronin/src/regldg/parse_regex.c:392: undefined reference to `log10'
parse_regex.o: In function `parse_regex_print_char_class':
/home/pcronin/src/regldg/parse_regex.c:1113: undefined reference to `log10'
parse_regex.o: In function `parse_regex_read_dec_val':
/home/pcronin/src/regldg/parse_regex.c:1438: undefined reference to `pow'
parse_regex.o: In function `parse_regex_pass_universe_specifier':
/home/pcronin/src/regldg/parse_regex.c:1261: undefined reference to `log10'
parse_regex.o: In function `parse_regex_scan':
/home/pcronin/src/regldg/parse_regex.c:241: undefined reference to `log10'
/home/pcronin/src/regldg/parse_regex.c:192: undefined reference to `log10'
collect2: error: ld returned 1 exit status
Makefile:21: recipe for target 'all' failed
make: *** [all] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ all: alt.o altlist.o build_structs.o char_set.o data.o debug.o \
 	$(LINK) alt.o altlist.o build_structs.o char_set.o data.o \
 	debug.o grouplist.o grouprecord.o main.o memory.o \
 	parse_regex.o program_args.o re_perm.o silist.o tnode.o \
-	vlr.o vlrlist.o -o regldg
+	vlr.o vlrlist.o -lm -o regldg
 alt.o: alt.h memory.h
 	$(COMPILE) alt.c -o alt.o
 altlist.o: altlist.h altlist.c alt.h memory.h

